### PR TITLE
Add `flyway` to migrate on the fly

### PR DESCRIPTION
**Pull request by pablomendezroyo**
Some migrations must be executed once the db is running, to do that: 
- Add a new service with `flyway` to perform those migrations on the fly. 
- Also, make the signer service dependant on this flyway service.
- Remove the migrations volume

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -19,40 +20,45 @@ services:
...
+      args:
+        UPSTREAM_VERSION: 23.3.0
+    depends_on:
+      postgres:
+        condition: service_started
+    image: "flyway.web3signer-prater.dnp.dappnode.eth:0.1.13"
+    restart: on-failure
   postgres:
     image: "postgres.web3signer-prater.dnp.dappnode.eth:0.1.13"
-    healthcheck:
```
[the remark this answers is not in the document]
Comment (pablomendezroyo) [reply docker-compose.yml:39]: Exactly! so the flow is:
1. postgres initializes
2. If postgres initializes fine, then flyway starts
3. If flyway sucessfully migrated, then signer starts
4. If Signer started then brain starts

File: flyway/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+##########
+# FLYWAY #
+##########
+FROM flyway/flyway:9.16.1-alpine
+ARG UPSTREAM_VERSION
+RUN apk update && apk add postgresql-client
+COPY --from=postgres-migrations /usr/src/app/web3signer-${UPSTREAM_VERSION}/slashing-protection/src/main/resources/migrations/postgresql/* /flyway/sql/
+COPY entrypoint.sh /usr/local/bin/entrypoint.sh
+RUN chmod +x /usr/local/bin/entrypoint.sh && rm /flyway/sql/put-your-sql-migrations-here.txt
```
[the remark this answers is not in the document]
Comment (pablomendezroyo) [reply flyway/Dockerfile]: Sounds good

File: flyway/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -e
+
+# Get postgresql database version and trim whitespaces
+DATABASE_VERSION=$(PGPASSWORD=password psql --tuples-only -U postgres -h postgres.web3signer-prater.dappnode -p 5432 -d web3signer -c "SELECT version FROM database_version WHERE id=1;" | awk '{print $1}' | tr -d '[:space:]')
```
[the remark this answers is not in the document]
Comment (pablomendezroyo) [reply flyway/entrypoint.sh:6]: > 

That would be the best solution, however if we define those envs in the compose file, they will appear as editable in the package config. And we do not want those envs to be editable

[the remark this answers is not in the document]
Comment (pablomendezroyo) [reply flyway/entrypoint.sh]: > Also, I think this approach is much cleaner:
> 
> `PGPASSWORD=password psql -U postgres -h postgres.web3signer-prater.dappnode -p 5432 -d web3signer -t -A -c "SELECT version FROM database_version WHERE id=1;"`

Sounds good

File: flyway/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+# Get the latest migration file version (ending in .sql) and trim whitespaces
+LATEST_MIGRATION_VERSION=$(ls -1 /flyway/sql/ | grep -E "V[0-9]+__.*.sql$" | tail -n 1 | cut -d'_' -f1 | cut -d'V' -f2 | sed 's/^0*//' | tr -d '[:space:]')
```
[the remark this answers is not in the document]
Comment (pablomendezroyo) [reply flyway/entrypoint.sh:9]: Not working as expected

**Review by dsimog01 — APPROVED** [review]

**Pull request merged**